mesonbuild/linkers/linkers.py

```diff
@@ -349,15 +349,25 @@ class VisualStudioLinker(VisualStudioLikeLinker, StaticLinker):
 
     """Microsoft's lib static linker."""
 
+    id = 'lib'
```

Comment:
In general regarding these ids, I think you're adding them for static linkers while currently we mostly do not implement self.id for static linkers at all, and do not seem to check this attribute or expose it anywhere particularly?

e.g. if I delete them all from this file in git master, the tests all pass.

This makes me inclined to believe it's okay to treat the addition of these ids as part of the same logical change as the validation logic you are adding.

It also means I'm not concerned about e.g. documenting these ids or caring about whether this will change what users see -- because users can only get access to cc.get_linker_id() providing the *dynamic* linker's identity.